ovos_utils/skills/templates/media_player.py

```diff
@@ -0,0 +1,151 @@
+    - will handle setting initial track status
+    - will fallback to audio only if GUI not connected
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if "download_audio" not in self.settings:
+            self.settings["download_audio"] = False
+        if "download_video" not in self.settings:
+            self.settings["download_video"] = False
+        if "audio_only" not in self.settings:
+            self.settings["audio_only"] = False
+        if "audio_with_video_stream" not in self.settings:
+            self.settings["audio_with_video_stream"] = False
+        if "mp3_audio" not in self.settings:
+            self.settings["mp3_audio"] = True
+        if "preferred_audio_backend" not in self.settings:
+            self.settings["preferred_audio_backend"] = None
+        self.message_namespace = basename(dirname(__file__)) + ".ovos_utils"
+        self.default_bg = "https://github.com/OpenVoiceOS/ovos_assets/raw/master/Logo/ovos-logo-512.png"
+        self.default_image = resolve_ovos_resource_file(
+            "ui/images/moviesandfilms.png")
+        self.bootstrap_list = []
+        if pyvod is None:
+            LOG.error("py_VOD not installed!")
+            LOG.info("pip install py_VOD>=0.4.0")
+            raise ImportError
+
+    def initialize(self):
+        self.add_event(
+            '{msg_base}.home'.format(msg_base=self.message_namespace),
+            self.handle_homescreen)
+        create_daemon(self.handle_bootstrap)
+
+    def handle_bootstrap(self):
+        # bootstrap, so data is cached
+        for url in self.bootstrap_list:
+            try:
+                if self.settings["download_audio"]:
+                    pyvod.utils.get_audio_stream(url, download=True,
+                                                 to_mp3=self.settings[
+                                                     "mp3_audio"])
+                if self.settings["download_video"]:
+                    pyvod.utils.get_video_stream(url, download=True)
+            except:
+                pass
+
+    # homescreen
+    def handle_homescreen(self, message):
+        # users are supposed to override this
+        self.CPS_start(self.name,
+                       {"image": self.default_image,
+                        "background": self.default_bg,
+                        "stream": random.choice(self.bootstrap_list)})
+
+    def CPS_match_query_phrase(self, phrase, media_type):
+        # users are supposed to override this
+        original = phrase
+        match = None
+
+        if match is not None:
+            return (phrase, match,
+                    {"media_type": media_type, "query": original,
+                     "image": self.default_image,
+                     "background": self.default_bg,
+                     "stream": random.choice(self.bootstrap_list)})
+        return None
+
+    def CPS_start(self, phrase, data):
+        self.play_media(data)
```

Comment:
Where's the other param going?

Reply:
in this case it's not used, but its part of the common play framework

Reply:
Alright. I'm flagging this "unread" so I remember to install and try it on a running instance.